src/Matrix.php

```diff
@@ -957,6 +957,110 @@ public function matmul(Matrix $b) : self
         return self::quick($c);
     }
 
+    /**
+     * Multiply this matrix with another matrix using the Strassen algorithm (matrix-matrix product).
+     *
+     * @param \Tensor\Matrix $b
+     * @throws \InvalidArgumentException
+     * @return self
+     */
+    public function matmulStrassen(Matrix $b) : self
+    {
+        if ($this->n !== $b->m()) {
+            throw new InvalidArgumentException('Matrix A requires'
+                . " $this->n rows but Matrix B has {$b->m}.");
+        }
+
+        if ($this->n === 1) {
+            return self::quick([[$this->a[0][0] * $b->a[0][0]]]);
+        }
+
+        if ($this->n % 2 !== 0) {
+            $a1 = $this->a;
+            $b1 = $b->a;
+
+            for ($i = 0; $i < $this->n + 1; ++$i) {
+                for ($j = 0; $j < $this->n + 1; ++$j) {
+                    $a1[$i][$j] = $a1[$i][$j] ?? 0;
```

Comment:
See other comment about doubly indexing, we've seen small performance enhancements by avoiding this type of indexing even on packed arrays. Our benchmarking suite can serve as a tool when optimizing this function. If you need help creating a benchmark, let me know I'll be happy to help.